### PR TITLE
DOC: Add contributing document

### PR DESCRIPTION
**Pull request by janbjorge**
Copy of https://github.com/equinor/xtgeo/blob/main/docs/contributing.rst with some modifications (dropped the section about how to run stuff in RMS)

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+Now you can make your changes locally.
+
+5. When you're done making changes, check that your changes pass ruff and the tests:
+
+.. code-block:: bash
+
+    $ ruff check .
+    $ pytest
+
+6. If you want to edit C code, take contact with the author for detailed instructions.
```
Comment (mferrera) [inline docs/contributing.rst:103]: Not a relevant note for dataio

File: docs/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+Standard acronyms to start the commit message with are:
+
+.. code-block:: text
+
+    API: an (incompatible) API change (will be rare)
+    BLD: change related to building fmu-dataio
+    BUG: bug fix
+    CLN: code cleanup, maintenance commit (refactoring, typos, PEP, etc.)
+    DEP: deprecate something, or remove a deprecated object
```
Comment (mferrera) [inline docs/contributing.rst:149]: Maybe @perolavsvendsen and @jcrivenaes have might feel differently, but I don't think we should bring numpy-style commit prefixing into this repo

File: docs/history.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+History
+========
```
Comment (mferrera) [inline docs/history.rst:2]: Should this be added? I think the commit history is usually good enough

**Review by mferrera — COMMENTED** [review]
@perolavsvendsen and @jcrivenaes will have the final word but I'm not sure we want to start adding the commit prefixing here too

File: docs/history.rst (hunk 3 above)
Comment (janbjorge) [reply docs/history.rst:2]: Not sure, wannted some place to track changes. But i think i want to use the "generate release notes" feature in GH insted.

Comment (janbjorge) [reply docs/history.rst]: Removed it.

File: docs/contributing.rst (hunk 2 above)
Comment (jcrivenaes) [reply docs/contributing.rst:147]: We have not made this convention for this repo (we have not actually discussed it either, yet).

**Review by jcrivenaes — CHANGES_REQUESTED** [review]
According to Equinor requirements, there shall be a CONTRIBUTING.md (MarkDown format) in top level. This is recently done in `xtgeo`, and then the MarkDown is linked into the `rst` documentation tree via `myst_parser`. See https://github.com/equinor/xtgeo/blob/main/docs/contributing.rst

I propope you should do it in the same manner here.

**Review by jcrivenaes — APPROVED** [review]
👍

**Pull request merged**